core/primitives/src/types.rs

```diff
@@ -120,14 +135,137 @@ impl SignedTransaction {
             nonce: 0,
             sender: AccountId::default(),
             receiver: AccountId::default(),
-            amount: 1,
-            method_name: String::new(),
+            amount: 0,
+            method_name: vec![],
             args: vec![],
         };
         SignedTransaction { sender_sig: DEFAULT_SIGNATURE, hash: hash_struct(&body), body }
     }
 }
 
+#[derive(Hash, Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum ReceiptBody {
+    NewCall(AsyncCall),
+    CallBack(CallBack),
+    Refund,
+}
+
+#[derive(Hash, Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AsyncCall {
+    pub amount: u64,
+    pub mana: u32,
+    pub method_name: Vec<u8>,
+    pub args: Vec<u8>,
+    pub callback: Option<CallBackId>,
+}
+
+impl AsyncCall {
+    pub fn new(amount: u64, mana: u32, method_name: Vec<u8>, args: Vec<u8>) -> Self {
```

Comment:
nit. Let's keep the order same as in the promise api: method_name, args, mana, amount

Reply:
Fixed in ca636d9